### PR TITLE
Use tinymarble.tif instead of the PNG

### PR DESCRIPTION
**Pull request by lnicola**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

We already have the TIFF in the repo, and this fixes the [block size](https://github.com/metasim/gdal/blob/0a50ba27b21ebe6dbe05d1ba1342dfa3c9af0fe1/src/raster/tests.rs#L507-L510) tests, where the official Docker images use `ENABLE_WHOLE_IMAGE_OPTIMIZATION`, which [causes](https://github.com/OSGeo/gdal/blob/bb931fe41aa2c348c6d40f4d3b0e3612246ba21e/frmts/png/pngdataset.cpp#L110-L119) the block size to be equal to the image size, while most distributions of GDAL don't.

I regenerated the existing TIFF, which has reversed (?), using:

```sh
gdal_translate tinymarble.png tinymarble.tif
gdal_edit.py -a_srs EPSG:4326 tinymarble.tif
gdaladdo -ro -r bilinear tinymarble.tif 2 4
```

### CODE REVIEW

File: src/metadata.rs [hunk 1]
```diff
@@ -320,26 +320,26 @@ mod tests {
...
     fn test_get_metadata_domains() {
-        let dataset = Dataset::open(fixture("tinymarble.png")).unwrap();
+        let dataset = Dataset::open(fixture("tinymarble.tif")).unwrap();
         let mut domains = dataset.metadata_domains();
-        if domains[0].is_empty() {
-            domains.remove(0);
-        }
+        domains.retain(|d| !d.is_empty());
+        domains.sort();
+        domains.dedup();
```
Comment (lnicola) [inline src/metadata.rs:327]: `ubuntu-lts` returns two `xml:XMP` entries.

File: src/raster/tests.rs [hunk 2]
```diff
@@ -759,17 +734,17 @@ fn test_raster_stats() {
     assert_eq!(
         rb.get_statistics(true, false).unwrap().unwrap(),
         StatisticsAll {
-            min: 12.0,
+            min: 0.0,
             max: 255.0,
-            mean: 89.2526,
-            std_dev: 90.99835379412092,
+            mean: 68.4716,
+            std_dev: 83.68444773934999,
```
Comment (metasim) [inline src/raster/tests.rs:740]: Any idea why the statistics changed? Is it computing approx stats via overviews?

**Comment by metasim** [comment]
bors d+

File: src/raster/tests.rs (hunk 2 above)
Comment (lnicola) [reply src/raster/tests.rs:740]: No, the original TIFF was somehow different from the PNG:

image

Reversed, on top, is the original TIFF, on bottom is the PNG converted to TIFF. Notice the darker area at the South pole and what looks like JPEG artifacts (but might be something else). I think both versions are derived from the real Blue Marble, but perhaps not in an ideal way.

**Comment by lnicola** [comment]
bors r=metasim

**Comment by lnicola** [comment]
bors retry

**Pull request merged**